submissions/VBystrov/friends-app/js/index.js

```diff
@@ -0,0 +1,337 @@
+const endpoint = 'https://randomuser.me/api/';
+const includeFields = `inc=${[
+  'gender',
+  'name',
+  'location',
+  'picture',
+  'dob',
+  'email',
+].join(',')}`;
+const amountFriends = `results=${50}`;
+const options = `?${[includeFields, amountFriends].join('&')}`;
+const requestFriendsUrl = `${endpoint}${options}`;
+const initialSearchParameters = {
+  name: null,
+  age: null,
+  gender: 'all',
+  sorting: 'nameAZ',
+};
+
+class FriendsApp {
+  constructor() {
+    this.init(requestFriendsUrl);
+  }
+
+  init(url) {
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((friendsData) => {
+        const friendlist = new Friendlist(friendsData.results);
+        const search = new Search(friendlist);
+      })
+      .catch((error) => {
+        console.error(
+          'There has been a problem with your fetch operation:',
+          error
+        );
+      });
+  }
+}
+
+class Friendlist {
+  constructor(friendsData) {
+    this.container = document.getElementById('friendlist');
+    this.createCards(friendsData);
+  }
+
+  createCards(friendlistData) {
+    this.cards = friendlistData.map((cardData) => new Card(cardData));
+  }
+
+  applySearch(searchParameters) {
+    this.applyFilters(searchParameters);
+    this.applySorting(searchParameters.sorting);
+  }
+
+  applyFilters(searchParameters) {
+    this.filteredCards = [...this.cards];
+
+    if (searchParameters.age) {
+      this.filteredCards = this.filteredCards.filter(
+        (card) => card.age === searchParameters.age
+      );
+    }
+
+    if (searchParameters.name) {
+      this.filteredCards = this.filteredCards.filter((card) =>
+        `${card.firstName} ${card.lastName}`
+          .toLowerCase()
+          .includes(searchParameters.name.toLowerCase())
+      );
+    }
+
+    if (searchParameters.gender !== 'all') {
+      this.filteredCards = this.filteredCards.filter(
+        (card) => card.gender === searchParameters.gender
+      );
+    }
+  }
+
+  applySorting(sortingOrder) {
+    let sortFunction;
+    switch (sortingOrder) {
+      case 'nameAZ':
+        const sortByNameAZ = function (card1, card2) {
+          const fullName1 = card1.firstName + card1.lastName;
+          const fullName2 = card2.firstName + card2.lastName;
+          if (fullName1 < fullName2) {
+            return -1;
+          }
+          if (fullName1 > fullName2) {
+            return 1;
+          }
+          if (fullName1 === fullName2) {
+            return 0;
+          }
+        };
```

Comment:
This function should be declared outside `applySorting`.

Separate responsibilities - `applySorting` should not know anything about the internals of the sorter function.